spec/ddtrace/opentracer/global_tracer_spec.rb

```diff
@@ -9,14 +9,14 @@
     describe '#global_tracer=' do
       subject(:global_tracer) { OpenTracing.global_tracer = tracer }
 
-      after { Datadog.configuration.tracer.instance = Datadog::Tracer.new }
+      after { Datadog::Tracing.configuration.tracer.instance = Datadog::Tracer.new }
```

Comment:
Should we deprecate this style of configuration setting? It ends up a being weird that the returned settings object can be directly mutated.

Reply:
Too significant a change to the API to make now.

It's currently really simple in that it passes the same object to `configure` as it returns from `configuration`. Now we're talking about returning an object that allows get behavior on all options, but not set, and returning a deprecation warning whenever set is accessed in a read-only context.

It'd be nice to have, but it's another feature to add to a configuration rework later.

Reply:
Could we just freeze the configuration object? E.g. treat this as separate from the proxies.